### PR TITLE
Implement third party notification support within alerts

### PR DESCRIPTION
**Pull request by relvira**
Added necessary structs to support third party notifications (via slack, pagerduty, etc) in alerts. This feature is lacking in your new-ish terraform provider and would be of great use. I used your API docs as reference: https://developer.thousandeyes.com/v6/alerts/#/alert-rule-create

I suppose a follow-up pull request on your provider using a newer version of this SDK will need to happen later. Let me know if anything else is needed and if I can facilitate any work to speed things up.

Thanks for your work. 🙇🏼

### CODE REVIEW

**Comment by sfreitas-te** [comment]
Thanks @relvira for your PR we will review as soon as possible

**Review by pedro-te — COMMENTED** [review]
Thank you very much for your contribution @relvira, this is much appreciated! 🙏🏻 
There are some small changes needed after testing your code, which I have written down as part of the review. 

Additionally, could you also add some tests to ensure the marshalling and unmarshalling of the new `NotificationThirdParty` struct works as expected?

Other than that, it looks good 👍🏻

**Comment by relvira** [comment]
Hey @pedro-te! 👋🏼 thanks for the review. I have updated the code with recommendations and added a test case for this, hope it all makes sense.

Let me know if there is anything else needed.

File: alert_test.go
Comment (pedro-te) [inline alert_test.go:109]:
```suggestion
```
Small nitpick here, the ThousandEyes API returns the value for `RuleID`, so in reality this field should not be populated here, but it should be populated in the `expected` variable, which you did. So this just needs to be removed.

**Review by pedro-te — COMMENTED** [review]
The rest looks good and it works well. Just another small ask: Can you squash the commits into a single one, whose message follows the _conventional commit_ format? So the commit message should say:

> feat: Implement third party notification support within alerts

Once this is updated we can merge the PR. Thanks again @relvira , your contribution is much appreciated

File: alert_test.go
Comment (relvira) [reply alert_test.go]: fixed the nit and squashed the commits @pedro-te. Thank you for reviewing this promptly. Once this is done I'll make a follow up contribution to your provider repo bumping the version of the sdk. 

Cheers! 🍻

File: alerts.go
Comment (pedro-te) [inline alerts.go:41]:
```suggestion
	IntegrationID   *string `json:"integrationId,omitempty"`
```

Ahh nice catch by the linter, this should be `IntegrationID` :)

**Review by pedro-te — CHANGES_REQUESTED** [review]
Apologies @relvira , this small detail slipped by.

**Comment by relvira** [comment]
Oops! good 👁️ - should be fixed and squashed now @pedro-te.

**Comment by pedro-te** [comment]
> Oops! good 👁️ - should be fixed and squashed now @pedro-te.

Just missing the tests now 😅 They're still using `IntegrationId`, we should be good to go after that 👍🏻

**Comment by relvira** [comment]
ah you're right. I think all should be good now, I've run tests locally and they are 🆗. Thanks again @pedro-te!

**Review by pedro-te — APPROVED** [review]
Thank you @relvira ! Looks good now!

**Review by sfreitas-te — APPROVED** [review]

**Comment by relvira** [comment]
Thank you @sfreitas-te and @pedro-te! will you generate a new release of the SDK with this included? I want to submit a PR to https://github.com/thousandeyes/terraform-provider-thousandeyes bumping the SDK version so we can use this feature in Terraform.

**Comment by pedro-te** [comment]
Yes, @relvira , we will!

**Comment by pedro-te** [comment]
@relvira v2.2.0 has been released! Apologies for the delay. Looking forward to the terraform provider PR :)

**Comment by relvira** [comment]
Wooho! appreciate the quick turnaround to create the release here, @pedro-te. I've just submitted a PR to the provider repo https://github.com/thousandeyes/terraform-provider-thousandeyes/pull/106. Feel free to 👀 when you have some time.

🙇🏼

**Pull request merged**
